From f97b057e70b66bcf502a4923f64d8b999b5814c3 Mon Sep 17 00:00:00 2001
From: Liang Yan <lyan@suse.com>
Date: Sat, 27 Oct 2018 11:07:10 -0400
Subject: [PATCH] Remove deprecated g_type_class_add_private()

The API has been deprecated, which causes build failures,
use the G_DEFINE_TYPE_WITH_PRIVATE() macro instead.

Signed-off-by: Liang Yan <lyan@suse.com>
---
 src/ldm.c | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/src/ldm.c b/src/ldm.c
index a55723c..1a9f2ec 100644
--- a/src/ldm.c
+++ b/src/ldm.c
@@ -271,7 +271,7 @@ struct _LDMPrivate
     GArray *disk_groups;
 };
 
-G_DEFINE_TYPE(LDM, ldm, G_TYPE_OBJECT)
+G_DEFINE_TYPE_WITH_PRIVATE(LDM, ldm, G_TYPE_OBJECT)
 
 static void
 ldm_dispose(GObject * const object)
@@ -296,7 +296,6 @@ ldm_class_init(LDMClass * const klass)
     GObjectClass *object_class = G_OBJECT_CLASS(klass);
     object_class->dispose = ldm_dispose;
 
-    g_type_class_add_private(klass, sizeof(LDMPrivate));
 }
 
 /* LDMDiskGroup */
@@ -322,7 +321,7 @@ struct _LDMDiskGroupPrivate
     GArray *comps;
 };
 
-G_DEFINE_TYPE(LDMDiskGroup, ldm_disk_group, G_TYPE_OBJECT)
+G_DEFINE_TYPE_WITH_PRIVATE(LDMDiskGroup, ldm_disk_group, G_TYPE_OBJECT)
 
 enum {
     PROP_LDM_DISK_GROUP_PROP0,
@@ -392,7 +391,6 @@ ldm_disk_group_class_init(LDMDiskGroupClass * const klass)
     object_class->finalize = ldm_disk_group_finalize;
     object_class->get_property = ldm_disk_group_get_property;
 
-    g_type_class_add_private(klass, sizeof(LDMDiskGroupPrivate));
 
     /**
      * LDMDiskGroup:guid:
@@ -487,7 +485,7 @@ struct _LDMVolumePrivate
     guint32 _n_comps_i;
 };
 
-G_DEFINE_TYPE(LDMVolume, ldm_volume, G_TYPE_OBJECT)
+G_DEFINE_TYPE_WITH_PRIVATE(LDMVolume, ldm_volume, G_TYPE_OBJECT)
 
 enum {
     PROP_LDM_VOLUME_PROP0,
@@ -573,7 +571,6 @@ ldm_volume_class_init(LDMVolumeClass * const klass)
     object_class->finalize = ldm_volume_finalize;
     object_class->get_property = ldm_volume_get_property;
 
-    g_type_class_add_private(klass, sizeof(LDMVolumePrivate));
 
     /**
      * LDMVolume:name:
@@ -731,7 +728,7 @@ struct _LDMPartitionPrivate
     LDMDisk *disk;
 };
 
-G_DEFINE_TYPE(LDMPartition, ldm_partition, G_TYPE_OBJECT)
+G_DEFINE_TYPE_WITH_PRIVATE(LDMPartition, ldm_partition, G_TYPE_OBJECT)
 
 enum {
     PROP_LDM_PARTITION_PROP0,
@@ -792,7 +789,6 @@ ldm_partition_class_init(LDMPartitionClass * const klass)
     object_class->finalize = ldm_partition_finalize;
     object_class->get_property = ldm_partition_get_property;
 
-    g_type_class_add_private(klass, sizeof(LDMPartitionPrivate));
 
     /**
      * LDMPartition:name:
@@ -866,7 +862,7 @@ struct _LDMDiskPrivate
     gchar *device; // NULL until device is found
 };
 
-G_DEFINE_TYPE(LDMDisk, ldm_disk, G_TYPE_OBJECT)
+G_DEFINE_TYPE_WITH_PRIVATE(LDMDisk, ldm_disk, G_TYPE_OBJECT)
 
 enum {
     PROP_LDM_DISK_PROP0,
@@ -944,7 +940,6 @@ ldm_disk_class_init(LDMDiskClass * const klass)
     object_class->finalize = ldm_disk_finalize;
     object_class->get_property = ldm_disk_get_property;
 
-    g_type_class_add_private(klass, sizeof(LDMDiskPrivate));
 
     /**
      * LDMDisk:name:
-- 
2.19.1

